heron/scheduler-core/src/java/com/twitter/heron/scheduler/SubmitterMain.java

```diff
@@ -374,9 +374,10 @@ public boolean submitTopology() {
 
       // create an instance of uploader
       uploader = ReflectionUtils.newInstance(uploaderClass);
+
     } catch (IllegalAccessException | InstantiationException | ClassNotFoundException e) {
-      LOG.log(Level.SEVERE, "Failed to instantiate instances", e);
-      return false;
+      throw new TopologySubmissionException(
```

Comment:
always better to attach the actual exception than to include it's message in the string.
```
throw new TopologySubmissionException("some message", e);
```
that gives the catcher more control to decide how to render it.